src/main/drivers/pwm_output.h

```diff
@@ -109,29 +109,36 @@ typedef enum {
 
 typedef struct {
     TIM_TypeDef *timer;
-    uint16_t timerDmaSources;
-#ifdef USE_DSHOT_DMAR
+#if defined(USE_DSHOT_DMAR)
+#if !defined(USE_HAL_DRIVER)
+    DMA_Stream_TypeDef *dmaBurstRef;
+    uint16_t dmaBurstLength;
+#endif
     uint32_t dmaBurstBuffer[DSHOT_DMA_BUFFER_SIZE * 4];
+#else
+    uint16_t timerDmaSources;
 #endif
 } motorDmaTimer_t;
 
 typedef struct {
     ioTag_t ioTag;
     const timerHardware_t *timerHardware;
     uint16_t value;
+#if !defined(USE_DSHOT_DMAR)
     uint16_t timerDmaSource;
+#endif
     motorDmaTimer_t *timer;
     volatile bool requestTelemetry;
 #if defined(STM32F3) || defined(STM32F4) || defined(STM32F7)
     uint32_t dmaBuffer[DSHOT_DMA_BUFFER_SIZE];
 #else
     uint8_t dmaBuffer[DSHOT_DMA_BUFFER_SIZE];
 #endif
-#if defined(STM32F7)
+#if defined(USE_HAL_DRIVER)
     TIM_HandleTypeDef TimHandle;
     DMA_HandleTypeDef hdma_tim;
     uint16_t timerDmaIndex;
-    uint8_t timerIndex;
+    uint8_t timerIndex; // XXX Will be gone when #4828 is merged
```

Comment:
Done now.

Reply:
Thanks! And also thanks for SPRACINGF4NEO thingy.